tests/test_proto_methods.rs

```diff
@@ -164,6 +168,16 @@ fn test_bool() {
     })
 }
 
+#[test]
+fn test_del() {
+    Python::with_gil(|py| {
+        let example_py = make_example(py);
+        example_py.setattr("special_custom_attr", 15).unwrap();
+        py_run!(py, example_py, "example_py.__del__()");
```

Comment:
Not sure what's the best way to test it.

> It looks like the tp_dealloc in that file also calls tp_finalize in some circumstances; we might need to check carefully that we reproduce the Python behaviour correctly.

@davidhewitt Could you elaborate on how do we check that?